scio-parquet/src/main/scala/com/spotify/scio/parquet/read/ParquetReadFn.scala

```diff
@@ -0,0 +1,237 @@
+package com.spotify.scio.parquet.read
+
+import com.spotify.scio.parquet.BeamInputFile
+import org.apache.beam.sdk.io.FileIO.ReadableFile
+import org.apache.beam.sdk.io.hadoop.SerializableConfiguration
+import org.apache.beam.sdk.io.range.OffsetRange
+import org.apache.beam.sdk.transforms.DoFn._
+import org.apache.beam.sdk.transforms._
+import org.apache.beam.sdk.transforms.splittabledofn.{OffsetRangeTracker, RestrictionTracker}
+import org.apache.beam.vendor.guava.v26_0_jre.com.google.common.collect.ImmutableSet
+import org.apache.parquet.filter2.compat.FilterCompat
+import org.apache.parquet.hadoop.ParquetFileReader
+import org.apache.parquet.hadoop.api.InitContext
+import org.apache.parquet.hadoop.metadata.BlockMetaData
+import org.apache.parquet.io.{ColumnIOFactory, ParquetDecodingException, RecordReader}
+import org.apache.parquet.{HadoopReadOptions, ParquetReadOptions}
+import org.slf4j.LoggerFactory
+
+import java.util.{Set => JSet}
+import scala.jdk.CollectionConverters._
+import scala.util.Try
+
+class ParquetReadFn[T, R](
```

Comment:
Why do we need a new SplittableDoFn implementation ? Beam already have https://beam.apache.org/releases/javadoc/2.36.0/org/apache/beam/sdk/io/parquet/ParquetIO.ReadFiles.html which seems to be fairly close.

Reply:
:+1: